plugins/rest/rest.go

```diff
@@ -67,21 +67,35 @@ func (c *Config) authPlugin(authPluginLookup func(string) HTTPAuthPlugin) (HTTPA
 	var candidate HTTPAuthPlugin
 	if c.Credentials.Plugin != nil && authPluginLookup != nil {
 		candidate := authPluginLookup(*c.Credentials.Plugin)
-		if candidate != nil {
+		if candidate == nil {
+			c.logger.WithFields(map[string]interface{}{
+				"plugin": *c.Credentials.Plugin,
+			}).Warn("Could not resolve candidate auth plugin")
```

Comment:
So, a config with an unknown plugin would cause a warning, but would otherwise "work"? It could also return an error, I suppose? 🤔  Or does it actually do that down the road... 🔍

Reply:
At the moment it is causing a fatal because fails the casting here: https://github.com/open-policy-agent/opa/pull/4175/files#diff-4effc5cf21474cce84e1a98bcda5492c1c44fdf2e93bb1bd29d719a7177883e1R96

We have two options here: 1. we fail correctly (return an error) or 2. not fail and log a warning. I could not find any mention in the original PR on why this was done this way.

Reply:
I also think (1.) is the better choice here. WDYT, @anderseknert?

Reply:
Agreed, as long as we are loud about it.

Reply:
Hej @anderseknert, so option 1 will be.